### PR TITLE
Update pedometer sample for jni 0.8.0

### PR DESCRIPTION
**Pull request by parlough**
Also clean up some nearby Dart code :)

### CODE REVIEW

**Comment by domesticmouse** [comment]
Looks like CI is unhappy about lack of super parameters

File: experimental/pedometer/example/lib/steps_repo.dart [hunk 1]
```diff
@@ -112,7 +114,7 @@ class _IOSStepsRepo implements StepsRepo {
           start, end, handler);
     }
 
-    return (await Future.wait(futures)).nonNulls.toList();
+    return (await futures.wait).nonNulls.toList();
```
[the remark this answers is not in the document]
Comment (parlough) [reply experimental/pedometer/example/lib/steps_repo.dart:117]: https://api.dart.dev/stable/3.3.3/dart-async/FutureIterable/wait.html

Generally should prefer this now, as it allows better error handling and is consistent with the extensions on records of futures.

**Review by HosseinYousefi — APPROVED** [review]
LGTM. Thanks @parlough!

**Comment by HosseinYousefi** [comment]
> Looks like CI is unhappy about lack of super parameters 

@parlough You can add an ignore comment in jnigen.yaml's preamble to fix this for now.

**Pull request merged**